evm_arithmetization/src/cpu/kernel/asm/transactions/type_0.asm

```diff
@@ -18,131 +18,96 @@ global process_type_0_txn:
     %stack (rlp_addr, len) -> (rlp_addr)
 
     // stack: rlp_addr, retdest
+    DUP1
+    // stack: rlp_addr, nonce_addr, retdest
```

Comment:
Nit: this got slightly confusing at first read, may be worth adding a one-liner comment here explaining we're saving the initial nonce position in the RLP segment? (technically the same applies for the other types of txns)